### PR TITLE
[KYUUBI #2025][HIVE] Add a Hive on Yarn doc 

### PR DESCRIPTION
**Pull request by deadwind4**
### _Why are the changes needed?_

jackson-annotations 2.13 and hive-exec 2.3.9 have class conflict

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: docs/deployment/engine_on_yarn.md [hunk 1]
```diff
@@ -179,3 +179,17 @@ $ echo "export HADOOP_CONF_DIR=/path/to/hadoop/conf" >> $KYUUBI_HOME/conf/kyuubi
 
 For experiment use, we recommend deploying Kyuubi Flink SQL engine in [Session Mode](https://nightlies.apache.org/flink/flink-docs-stable/docs/deployment/resource-providers/yarn/#session-mode).
 At present, [Application Mode](https://nightlies.apache.org/flink/flink-docs-stable/docs/deployment/resource-providers/yarn/#application-mode) and [Per-Job Mode (deprecated)](https://nightlies.apache.org/flink/flink-docs-stable/docs/deployment/resource-providers/yarn/#per-job-mode-deprecated) are not supported for Flink engine.
+
+## Deploy Kyuubi Hive Engine on Yarn
+
+### Requirements
+
+When you want to deploy Kyuubi's Flink SQL engines on YARN, you'd better have cognition upon the following things.
```
Comment (yaooqinn) [inline docs/deployment/engine_on_yarn.md:187]: seems that content is all about flink?

Comment (deadwind4) [reply docs/deployment/engine_on_yarn.md:187]: Sorry, I haven't done it. The PR is a draft. coming soon.

**Comment by yaooqinn** [comment]
shall we also mention how to configure hive-site.xml on the hive engine side?

**Comment by deadwind4** [comment]
> shall we also mention how to configure hive-site.xml on the hive engine side?

Nothing needs to mention about hive-site.xml, because we have little mention of Yarn options like Flink on Yarn docs, we assume users have read Spark/Flink/Hive official docs. And many Yarn-related options are about MapReduce/Tez not in `hive-site.xml`.

In the future, we can make a Q&A doc about Spark/Flink/Hive their self-configuration after we collect feedback questions in the Kyuubi community.
@yaooqinn

**Review by yaooqinn — APPROVED** [review]

**Pull request closed without merging**